Tools/cases_generator/generate_cases.py

```diff
@@ -1257,11 +1281,36 @@ def always_exits(lines: list[str]) -> bool:
     )
 
 
-def variable_used(node: parser.Node, name: str) -> bool:
-    """Determine whether a variable with a given name is used in a node."""
-    return any(
-        token.kind == "IDENTIFIER" and token.text == name for token in node.tokens
-    )
+def variable_used(node: parser.Node, name: str, delim: str = None) -> bool:
+    """Determine whether a variable with a given name is used in a node.
+       If delim is not None, the name is assumed to be a chain of names
+       delimited by delim. For example: "frame->f_code->co_consts", delim="->".
+    """
+    if delim is None:
+        return any(
+            token.kind == "IDENTIFIER" and token.text == name for token in node.tokens
+        )
+
+    # delim is not None - look for the sequence of names
+    orig_names = name.split(delim)
+    orig_names.reverse()
+    names = orig_names
+
+    need_delim = False
+    for token in node.tokens:
+        if need_delim:
+            if token.text == delim:
+                need_delim = False
+            else:
+                # no match, start over
+                names = orig_names
+                continue
+        if token.kind == "IDENTIFIER" and token.text == names[-1]:
+            names.pop()
```

Comment:
This looks like it would mutate `orig_names`.

In general this looks too clever. Here's how I would do it:
- Convert "foo->bar->baz" into a list of pairs
  `[("IDENTIFIER", "foo"), (None, "->"), ("IDENTIFIER", "bar"), (None, "->"), ("IDENTIFIER", "baz")]`
- Try to match the `(type, text)` of input tokens to those pairs, ignoring the type if the pair has None

Reply:
I wonder if we should instead add to the DSL names for these fields (C macros mapping the name to the sequence). Then we don't need this in the generator, we just look for the name as a single identifier.